### PR TITLE
Reverse chron

### PR DESCRIPTION
**Pull request by mejango**
Attempt at storing and returning tiers in reverse chron order of being added, within categories.

needs test, and likely some logic edge case touchups. 

spec: 

Create with tiers 
[
{id: 1, cat: 1}, 
{ id 2, cat: 2}, 
{ id: 3, cat 2}, 
{id: 4, cat: 3}
]

Add new tiers 
[
{id: 5, cat 1}, 
{id: 6, cat 1},
{id: 7: cat 2}, 
{ id: 8, cat 4}
] 

should give me full sorted order:

[
{id: 5, cat 1}, (new)
{id: 6, cat 1}, (new)
{id: 1, cat: 1}, 
{id: 7: cat 2}, (new)
{ id 2, cat: 2}, 
{ id: 3, cat 2}, 
{id: 4, cat: 3}.
{ id: 8, cat 4} (new)
]

reasoning: makes more sense from a viewing perspective to get the most recent stuff first.

### CODE REVIEW

**Review by drgorillamd — APPROVED** [review]

File: contracts/forge-test/NFTReward_Unit.t.sol [hunk 1]
```diff
@@ -2214,6 +2214,319 @@ contract TestJBTieredNFTRewardDelegate is Test {
...
+        category: _tierParamsToAdd[i].category,
+        allowManualMint: _tierParamsToAdd[i].allowManualMint,
+        transfersPausable: _tierParamsToAdd[i].transfersPausable
+      });
+      vm.expectEmit(true, true, true, true, address(_delegate));
+      emit AddTier(_tiersAdded[i].id, _tierParamsToAdd[i], owner);
+    }
+
+    vm.startPrank(owner);
+    _delegate.adjustTiers(_tierParamsToAdd, new uint256[](0));
```
Comment (drgorillamd) [inline contracts/forge-test/NFTReward_Unit.t.sol:2336]: Shouldn't we remove some too?

Comment (mejango) [reply contracts/forge-test/NFTReward_Unit.t.sol:2336]: yes plz @viraj124

Comment (viraj124) [reply contracts/forge-test/NFTReward_Unit.t.sol:2338]: pushed
